train.py

```diff
@@ -325,4 +340,36 @@ def train_func (data_path, existing_model, model_path, BATCH_SIZE, visualize_dat
             confusion = confusion_matrix(valid_labels_flat, valid_preds_flat)
             # Print or use the confusion matrix as needed
             print("Confusion Matrix:")
-            print(confusion)
\ No newline at end of file
+            print(confusion)
+
+            
+            # Create a DataFrame for better visualization
+            df_cm = pd.DataFrame(confusion, index=classes_name, columns=classes_name)
+                    
+            # Plot the confusion matrix
+            plt.figure(figsize=(10, 7))
+            sn.heatmap(df_cm, annot=True, fmt='d', cmap="crest")
+            plt.title("Confusion Matrix")
+            plt.xlabel("Predicted")
+            plt.ylabel("True")
+            confusion_matrix_path = os.path.join(os.path.dirname(model_path), "confusion_matrix.png")
+            plt.savefig(confusion_matrix_path)
+            plt.show()
+            
+            # Generate and print classification report
+            class_report = classification_report(valid_labels_flat, valid_preds_flat, target_names=classes_name, output_dict=True)
+            df_class_report = pd.DataFrame(class_report).transpose()
+            
+            # Plot classification report
```

Comment:
Did you find out whether this classification report is the same as the confusion matrix? Also, does it solve the problem with classes being unidentifiable in case not all classes are predicted? I remember you once said that the implementation you were using has an issue with showing less classes in case not all are predicted and then its not possible to trace back which class labels refer to which class.

Reply:
No, if the number of actual classes matches the number of predicted classes (as it should in an ideal situation), everything will work correctly. However, if the numbers differ, an error message will be generated, indicating that the number of true labels (number) does not equal the number of predicted labels (number).
But yea I will try to edit this process where we can compare between input classes and output to manage that in the next branch :)